### PR TITLE
Endpoint to fetch a todo item added

### PR DESCRIPTION
**Pull request by aniketrathi**
## Description
Fetch a TODO item:
URI: GET /todos/:id
Response HTTP Status: 200
Response body: { id: id, title: title }

## Database schema changes
- None

### Automated test cases added
- Test in which we get Response 200 if todo item exists else 404

### Manual test cases run
- None

### CODE REVIEW

File: src/controllers/todo-controller.ts [hunk 1]
```diff
@@ -79,4 +81,15 @@ export class TodoController extends BaseController {
...
   };
+
+  private fetchTodo = async (req: ExtendedRequest, res: Response, next: NextFunction) => {
+    const { id } = req.params;
+    const todo = await this.appContext.todoRepository.findById(id);
+
+    if (todo?._id) {
+      res.status(200).json(todo.serialize());
+    } else {
+      res.status(404).send();
```
Comment (jjalan) [inline src/controllers/todo-controller.ts]: We should have a generic error for 404 like for other cases.

File: test/spec/todo/todo.spec.ts [hunk 2]
```diff
@@ -120,3 +120,18 @@ describe("PUT /todos/:id", () => {
     }
   });
 });
+
+describe("GET /todos/:id", () => {
+  it("should return 200 if todo exists else 404", async () => {
```
Comment (jjalan) [inline test/spec/todo/todo.spec.ts]: This is very poor test case.

A test case should only test one thing.

Comment (aniketrathi) [reply test/spec/todo/todo.spec.ts]: done sir

File: src/controllers/todo-controller.ts (hunk 1 above)
Comment (aniketrathi) [reply src/controllers/todo-controller.ts]: corrected

**Pull request merged**